packages/discovery-provider/ddl/functions/handle_manager_request.sql

```diff
@@ -0,0 +1,67 @@
+create or replace function process_grant_change() returns trigger as $$
+declare
+    matched_user_id integer;
+begin
+    -- fetch the user_id where wallet matches grantee_address
+    select user_id into matched_user_id from users where lower(wallet) = lower(new.grantee_address);
+    
+    if matched_user_id is not null then
+        if (tg_op = 'insert' and new.is_revoked = false and new.is_approved is null and new.created_at = new.updated_at or
```

Comment:
Mind adding a comment or two to this function to describe what's happening in each case?
There are a lot of ands/ors on this line :-D

Reply:
will do!